python/lsst/rubintv/production/slac/mosaicing.py

```diff
@@ -259,7 +277,22 @@ def makeMosaic(deferredDatasetRefs, camera, binSize, dataPath, timeout, deleteAf
             time.sleep(0.5)
             continue
 
+    if success and deleteIfComplete and (len(detectorNameList) == nExpected):
+        # Remaking the image just to delete the files is pretty gross, but it's
```

Comment:
I think it should well do actually (though whether that would actually be much faster or less gross is a different question). I'm going to punt on this very temporarily though, because I've already started DM-39698 and that's going to try to tackle improving file deletion again, so will hopefully address this in that too.